examples/policies-with-custom-data-sources/README.md

```diff
@@ -0,0 +1,213 @@
+# Policies with custom data sources
+
+This example creates a policy via terraforms [`aws_iam_policy_document`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) and then adds it to a role's `policy_arns` list.
+
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [Example](#example)
+    - [Variable definition](#variable-definition)
+    - [Use data source to fetch a dynamic value](#use-data-source-to-fetch-a-dynamic-value)
+    - [Build our policy](#build-our-policy)
+    - [Enrich roles list with created policy](#enrich-roles-list-with-created-policy)
+    - [Define the iam module](#define-the-iam-module)
+5. [Usage](#usage)
+6. [Requirements](#requirements)
+7. [Providers](#providers)
+8. [Inputs](#inputs)
+9. [Outputs](#outputs)
+
+
+## Overview
+
+By using [`aws_iam_policy_document`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) you have the advantage to use terraform's variables inside the policy definition and be able to first gather already existing resources and add them to that policy. This makes the process of creating policies as dynamic and flexible as it can get.
+
+
+## Example
+
+The following defines two roles, both without any policy attached.
+
+#### Variable definition
+
+`terraform.tfvars`
+```hcl
+roles = [
+  {
+    name                 = "ROLE-ADMIN"
+    path                 = null
+    desc                 = null
+    trust_policy_file    = "data/trust-policy-file.json"
+    permissions_boundary = null
+    policies             = []
+    inline_policies      = []
+    policy_arns          = []
+  },
+  {
+    name                 = "ROLE-DEV"
+    path                 = null
+    desc                 = null
+    trust_policy_file    = "data/trust-policy-file.json"
+    permissions_boundary = null
+    policies             = []
+    inline_policies      = []
+    policy_arns          = []
+  },
+]
+```
+
+#### Use data source to fetch a dynamic value
+
+Now we're going to dynamically fetch the current AWS account id (just for the sake of this example to have something dynamic).
+
+`main.tf`
+```hcl
+data "aws_caller_identity" "current" {}
+```
+
+#### Build our policy
+
+We can then use this account id and include it in our policy document.
+
+`main.tf`
+```hcl
+data "aws_iam_policy_document" "s3" {
+  statement {
+    sid = "1"
+
+    actions = [
+      "s3:ListAllMyBuckets",
+    ]
+
+    resources = [
+      "arn:aws:s3::${data.aws_caller_identity.current.account_id}:*",
+    ]
+  }
+}
+```
+
+Based on the created policy document, we can define our policy.
+
+`main.tf`
+```hcl
+resource "aws_iam_policy" "s3" {
+  name        = "s3-policy"
+  path        = "/custom/"
+  description = "Custom S3 policy"
+  policy      = data.aws_iam_policy_document.s3.json
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+```
+
+#### Enrich roles list with created policy
```

Comment:
Is this your take for the statement overriding and if so, how does it compare to the [built-in functionality](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document#example-with-source-and-override)?

Reply:
No, this is not another overriding take. This shows simply how to use a data source. Then for the data source you can either use it normally or use the built-in oveerride functionality.

Reply:
Not exactly, as there `override_json` is not exposed anywhere but you answered my question.